### PR TITLE
Add simple report generator based on ASV results

### PR DESCRIPTION
**Pull request by densmirn**

### CODE REVIEW

File: tests_perf/asvgen.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                machine_info = json.load(fd)
+            for res_path in subdir.glob('*.json'):
+                if res_path.name == self.machine_json:
+                    # Skip machine info file
+                    continue
+
+                with res_path.open(encoding='utf-8') as fd:
+                    results = json.load(fd)['results']
+                data = {}
+                for benchmark, result in results.items():
```
Comment (fschlimb) [inline tests_perf/asvgen.py]: A comment briefly explaining the following line wouldn't hurt...

**Comment by fschlimb** [comment]
@densmirn This looks good.
Could you send me a sample output html file?

**Comment by densmirn** [comment]
Row # | Benchmark | Result, s | Min, s | Arithmetic Mean, s | Max, s | Standard Deviation, s
-- | -- | -- | -- | -- | -- | --
1 | strings.Methods.time_len('compiled_python') | 1.8699214337393641 | 1.8492342680692673 | 1.8806827198714018 | 1.9336537439376116 | 0.03205774126323404
2 | strings.Methods.time_len('interpreted_python') | 1.30884275957942 | 1.2758314218372107 | 1.3905458934605122 | 1.5869634989649057 | 0.13954054110769443
3 | strings.Methods.time_capitalize('interpreted_python') | 3.4665139112621546 | 3.4602694008499384 | 3.4665139112621546 | 3.4727584216743708 | 0.0062445104122161865
4 | strings.Methods.time_capitalize('compiled_python') | 20.32579606771469 | 19.61896459199488 | 20.32579606771469 | 21.0326275434345 | 0.7068314757198095
5 | strings.Methods.time_lower('interpreted_python') | 2.1186749674379826 | 2.0423572938889265 | 2.1186749674379826 | 2.1949926409870386 | 0.07631767354905605
6 | strings.Methods.time_lower('compiled_python') | 19.162730019539595 | 18.21585246361792 | 19.162730019539595 | 20.10960757546127 | 0.9468775559216738
7 | strings.Methods.time_swapcase('interpreted_python') | 5.9405054254457355 | 5.86518438346684 | 5.9405054254457355 | 6.015826467424631 | 0.07532104197889566
8 | strings.Methods.time_swapcase('compiled_python') | 22.8039807388559 | 22.078664841130376 | 22.8039807388559 | 23.52929663658142 | 0.7253158977255225
9 | strings.Methods.time_title('interpreted_python') | 4.401013186201453 | 4.399192996323109 | 4.401013186201453 | 4.402833376079798 | 0.0018201898783445358
10 | strings.Methods.time_title('compiled_python') | 21.98942913673818 | 21.93336272984743 | 21.98942913673818 | 22.04549554362893 | 0.05606640689074993
11 | strings.Methods.time_upper('interpreted_python') | 2.126695491373539 | 2.112878108397126 | 2.126695491373539 | 2.1405128743499517 | 0.013817382976412773
12 | strings.Methods.time_upper('compiled_python') | 18.43956960644573 | 18.36489922180772 | 18.43956960644573 | 18.51423999108374 | 0.07467038463801146

**Comment by fschlimb** [comment]
I suggest using bokeh or seaborn or matplotlib or alike. Even a simple boxplot is much easier to read than a table and super simple to setup with these tools.

**Pull request merged**